src/components/modal/index.tsx

```diff
@@ -35,22 +35,25 @@ const Modal: FC<ModalProps> = ({
   const overflowClass = verticalOverflow
     ? `overflow-y-${verticalOverflow}`
     : "md:overflow-y-auto overflow-y-scroll"
-  const handleClose = () => {
+  const handleClose = useCallback(() => {
     onClose ? onClose() : null
     close()
-  }
+  }, [onClose, close])
```

Comment:
did you make sure that every component that uses the modal and passes a `onClose` function wraps it into a callback? because otherwise this function is recreated on every change of the parent. Thus triggers a change on handleClose, then handleKeys and at last the useEffect

Reply:
For simple components, I saw that it's fine to let the component rerender because in most cases its a lighter operation than keeping a cache / checking against it.  That's basically the reason we shoudn't wrap every function with `useCallback`

Reply:
if the `onClose` will always have a different ref then we dont need `useCallback`

Reply:
as discussed, we will disable the rule for this component